css-view-transitions-1/Overview.bs

```diff
@@ -1424,19 +1437,20 @@ urlPrefix: https://html.spec.whatwg.org/multipage/rendering.html; type: dfn;
 
 		1. [=Assert=]: |transition|'s [=ViewTransition/phase=] is [=phases/before=] "`update-callback-called`".
 
-		1. Let |callbackPromise| be [=a new promise=] in |transition|'s [=relevant Realm=].
+		1. Let |callbackPromise| be null.
 
-			* If |transition|'s [=ViewTransition/update callback=] is null, then resolve |callbackPromise|.
+		1. If |transition|'s [=ViewTransition/update callback=] is null,
+			then set |callbackPromise| to [=a promise resolved with=] undefined,
+			in |transition|'s [=relevant Realm=].
 
-			* Otherwise, let |callbackPromise| be the result of [=/invoking=] |transition|'s [=ViewTransition/update callback=].
+		1. Otherwise, set |callbackPromise| to the result of [=/invoking=] |transition|'s [=ViewTransition/update callback=].
 
 		1. Set |transition|'s [=ViewTransition/phase=] to "`update-callback-called`".
 
-		1. [=promise/React=] to |callbackPromise|:
-
-			* If |callbackPromise| was resolved, then [=resolve=] |transition|'s [=ViewTransition/update callback done promise=].
+		1. [=Resolve=] |transition|'s [=ViewTransition/update callback done promise=]
```

Comment:
I feel like this is an extremely subtle point which I wouldn't have known if you didn't link to the your 2014 article.

"resolve the promise with the result of reacting to callbackPromise". This does, in fact, mean reject the promise if callbackPromise was rejected, right?

I'm not sure what race conditions you're referencing below, but I think it would be better to be explicit here "reject if rejects, resolve if resolved" type of language

Reply:
> "resolve the promise with the result of reacting to callbackPromise". This does, in fact, mean reject the promise if callbackPromise was rejected, right?

Yeah, the process of [react](https://webidl.spec.whatwg.org/#dfn-perform-steps-once-promise-is-settled) is very similar to `promise.then()`, as in, if you don't react to the reject case, then the rejection is passed through.

I'll add a note to make this clear.

The only reason the fulfilled case is handled is to ensure that `updateCallbackDone` fulfills with `undefined` rather than whatever the callback returns.

Reply:
> I'm not sure what race conditions you're referencing below

We didn't have any in the spec right now, but since our intention was to have `updateCallbackDone` follow `callbackPromise`, except resolving with `undefined`, it seems better to spec it with that intent.

Here's the difference between the two patterns: https://jsbin.com/butulov/edit?js,console